src/Map/ProspectorOverlayLayer.cs

```diff
@@ -343,13 +449,24 @@ public void RebuildMap(bool rebuildTexture = false)
 
             if (rebuildTexture)
             {
-                _colorTexture?.Dispose();
-                _colorTexture = GenerateOverlayTexture();
+                for (int i = 0; i < _colorTextures.Length; i++)
+                {
+                    _colorTextures[i]?.Dispose();
+                    _colorTextures[i] = GenerateOverlayTexture((RelativeDensity)i);
+                }
             }
 
-            foreach (var message in _messages)
+            foreach (var message in _prospectInfos)
             {
-                var component = new ProspectorOverlayMapComponent(_clientApi, message.X, message.Z, message.Message, _colorTexture);
+                RelativeDensity densityValue;
+                if (_config.HeatMapOre == null)
```

Comment:
Maybe extract determining the density to a helper method as it is needed here and when parsing chat messages